zedd-app/src/AppState.ts

```diff
@@ -539,11 +539,28 @@ export class AppState {
   )
 
   public fillErsatz(when: Interval): void {
-    for (const day of eachDayOfInterval(when)) {
-      this.addSliceIfDayEmpty(
-        this.makeFullDaySlice(day, this.getTaskForName(this.config.ersatzTask)),
-      )
+    this.getHolidays((when.start as Date).getFullYear()).then((holidays) => {
+      for (const day of eachDayOfInterval(when)) {
+        let task = this.getTaskForName(this.config.ersatzTask)
+        if (this.config.holidayClarityTaskIntId && holidays.length > 0) {
+          holidays.forEach((holiday: { date: string }) => {
+            if (day.toLocaleDateString() === new Date(holiday.date).toLocaleDateString()) {
+              task = new Task('HOLIDAY', this.config.holidayClarityTaskIntId)
```

Comment:
Creating multiple tasks with the same name will probably lead to errors. Can we use getTaskForName as above?